inference-engine/src/gna_plugin/transformations/decompose_2d_conv.cpp

```diff
@@ -0,0 +1,597 @@
+    size_t total_factorized_conv_channel_count = (conv_data.input_channel_count * conv_data.filter_height * conv_data.filter_width);
+    while (total_factorized_conv_channel_count / graph_data.conv_count > GNALimitations::convFilterMaxSize ||
+        total_factorized_conv_channel_count % graph_data.conv_count != 0 || conv_data.filter_channel_count % graph_data.conv_count != 0)
+        graph_data.conv_count++;
+
+    // Concat (copy) layer limitation allows to split up to a certain limit
+    // Currently we are able to split only convolutions without pooling in horizontal dimension
+    if (graph_data.conv_count > GNALimitations::copyMaxGrouping ||
+        ((graph_data.pool_size_width > 1 || graph_data.pool_stride_width > 1) && graph_data.conv_count > 1))
+        return false;
+
+    // GNA supported features - there is no need to decompose such convolution
+    if (graph_data.conv_count == 1 && (conv_data.input_height == 1 || conv_data.input_width == 1) &&
+        conv_data.filter_dilation_width == 1 && conv_data.filter_dilation_height == 1)
+        return false;
+
+    return true;
+}
+
+template<typename T>
+static std::vector<std::shared_ptr<ngraph::opset7::Constant>> Split2DConvFilters(std::shared_ptr<ngraph::opset7::Constant>& filters,
+    const bool& vertical_permute, const bool& horizontal_permute, const ngraph::element::Type& element_type, const size_t& split_channels) {
+
+    std::vector <std::shared_ptr<ngraph::opset7::Constant>> result;
+    auto filter_shape = filters->get_output_shape(0);
+    if (!horizontal_permute && !vertical_permute && split_channels == 1)
+        return {filters};
+
+    IE_ASSERT(filter_shape.size() == 4);
+
+    std::vector<std::vector<float>> flat_filters;
+    flat_filters.resize(split_channels);
+    for (size_t i = 0; i < split_channels; i++)
+        flat_filters[i].resize(shape_size(filter_shape) / split_channels);
+
+    auto N = filter_shape[0];
+    auto C = filter_shape[1];
+    auto H = filter_shape[2];
+    auto W = filter_shape[3];
+
+    size_t CS = (C / split_channels);
+    const auto* data = filters->get_data_ptr<T>();
+
+    for (size_t n = 0; n < N; n++) {
+        for (size_t c = 0; c < CS; c++) {
+            for (size_t s = 0; s < split_channels; s++) {
+                for (size_t h = 0; h < H; h++) {
+                    for (size_t w = 0; w < W; w++) {
+                        if (vertical_permute || !horizontal_permute) {
+                            flat_filters[s][n * CS * H * W + c * H * W + h * W + w] =
+                                data[n * C * H * W + (c * split_channels + s) * H * W + h * W + w];
+                        } else if (horizontal_permute) {
+                            flat_filters[s][n * CS * H * W + c * H * W + H * w + h] =
+                                data[n * C * H * W + (c * split_channels + s) * H * W + h * W + w];
+                        }
+                    }
+                }
+            }
+        }
+    }
```

Comment:
The same question actually: why we can't do it using ngraph node split and constant folding?